pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -214,6 +241,40 @@ func (r *ReconcileBareMetalHost) Reconcile(request reconcile.Request) (reconcile
 		return reconcile.Result{Requeue: true}, nil
 	}
 
+	// Set the hardware profile name.
+	if host.Status.HardwareDetails == nil {
+		if dirty, err = r.provisioner.InspectHardware(host); err != nil {
+			reqLogger.Error(err, "failed to inspect hardware")
+			return reconcile.Result{}, err
+		}
+		if dirty {
+			reqLogger.Info("saving status after inspecting hardware")
+			if err := r.client.Update(context.TODO(), host); err != nil {
+				reqLogger.Error(err, "failed to save host during inspection")
+				return reconcile.Result{}, err
+			}
+			if host.Status.HardwareDetails != nil {
+				reqLogger.Info("saving hardware details")
```

Comment:
I suggest using similar language to the error that comes two lines below, so it would be clear to someone looking in a log that this is the operation that failed. "Doing X" ... "Failed to do X"

Reply:
ah, yeah, some of these messages got moved around. I'll clean all of that up.